my/core/__main__.py

```diff
@@ -542,6 +542,14 @@ def query_hpi_functions(
                 pprint(item)
         else:
             pprint(list(res))
+    elif output == 'gpx':
+        from my.location.common import locations_to_gpx
+
+        # can ignore the mypy warning here, locations_to_gpx yields any errors
+        # if you didnt pass it something that matches the LocationProtocol
+        for exc in locations_to_gpx(res, sys.stdout):   # type: ignore[arg-type]
```

Comment:
I guess one downside is that it doesn't respect the same exception policies as select_range, and just dumps everything to terminal. But not a big deal, and perhaps can think about it later if necessary

Reply:
ah yeah, this was more just a safeguard to prevent crashing, can add the two if statements to handle errors

Reply:
one (somewhat hacky?) way to make it fit into the existing mechanism might be hacking a wrapper iterable before calling `select_range`, e.g.
```
if output == 'gpx':
    input_src = map(check_conforms_to_location_procotol, input_src)

# select_range call etc
```

Reply:
Ah I see what you mean -- that way exceptions will end up in stdout and mess with gpx output. Anyway I think it's fine as it is :) But yeah later might worth adding an option to `select_range`/cli support to log exceptions to stderr?

Reply:
yeah sounds good, will do that when I PR some docs for `hpi query`